tests/worker.test.ts

```diff
@@ -0,0 +1,38 @@
+import { describe, it, expect, beforeAll, afterAll } from 'vitest';
+import { AgentRunner } from '../worker/runner';
+import { AGENT_SERVER_PORT } from '../server/constants';
+
+describe('Worker Health Check', () => {
+  let agentRunner: AgentRunner;
+  const testPort = AGENT_SERVER_PORT + 1000;
+
+  beforeAll(async () => {
+    process.env.HEALTH_PORT = testPort.toString();
+    process.env.LOG_LEVEL = 'error';
+    
+    agentRunner = new AgentRunner({ 
+      sleepTime: 0.1,
+      maxLoops: 1
+    });
+    
+    await agentRunner.run();
+    
+    await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
Instead of waiting 1sec, update the agent runner await statement such that the method resolves once the port is ready